src/main/kotlin/com/thelastpickle/tlpstress/profiles/udttimeseries/UdtTimeSeries.kt

```diff
@@ -0,0 +1,90 @@
+package com.thelastpickle.tlpstress.profiles.udttimeseries
+
+import com.datastax.driver.core.PreparedStatement
+import com.datastax.driver.core.Session
+import com.datastax.driver.core.utils.UUIDs
+import com.thelastpickle.tlpstress.PartitionKey
+import com.thelastpickle.tlpstress.StressContext
+import com.thelastpickle.tlpstress.generators.*
+import com.thelastpickle.tlpstress.profiles.IStressProfile
+import com.thelastpickle.tlpstress.profiles.IStressRunner
+import com.thelastpickle.tlpstress.profiles.Operation
+import java.util.UUID
+
+
+
+
+/**
+ * Create a simple time series use case with some number of partitions
+ * TODO make it use TWCS
+ */
+class UdtTimeSeries : IStressProfile {
+
+    data class PrimaryKey(val first: String, val timestamp: UUID)
+
+    override fun schema(): List<String> {
+        val queryUdt = """CREATE TYPE IF NOT EXISTS sensor_data_details (
+                          data1 text,
+                          data2 text,
+                          data3 text
+                        );
+                           """.trimIndent()
+
+        val queryTable = """CREATE TABLE IF NOT EXISTS sensor_data_udt (
+                            sensor_id text,
+                            timestamp timeuuid,
+                            data frozen<sensor_data_details>,
+                            primary key(sensor_id, timestamp))
+                            WITH CLUSTERING ORDER BY (timestamp DESC)
+                           """.trimIndent()
+
+        return listOf(queryUdt, queryTable)
+    }
+
+    lateinit var prepared: PreparedStatement
+    lateinit var getRow: PreparedStatement
+    lateinit var getPartitionHead: PreparedStatement
+
+    override fun prepare(session: Session) {
+        prepared = session.prepare("INSERT INTO sensor_data_udt (sensor_id, timestamp, data) VALUES (?, ?, ?)")
+        getRow = session.prepare("SELECT * from sensor_data_udt WHERE sensor_id = ? AND timestamp = ? ")
+        getPartitionHead = session.prepare("SELECT * from sensor_data_udt WHERE sensor_id = ? LIMIT ?")
+    }
+
+    /**
+     * need to fix custom arguments
+     */
+    override fun getRunner(context: StressContext): IStressRunner {
+
+        val dataField = context.registry.getGenerator("sensor_data", "data")
+
+        class TimeSeriesRunner(val insert: PreparedStatement, val select: PreparedStatement, val limit: Int) : IStressRunner {
+
+            override fun getNextSelect(partitionKey: PartitionKey): Operation {
+
+                val bound = select.bind(partitionKey.getText(), limit)
+                return Operation.SelectStatement(bound)
+            }
+
+            override fun getNextMutation(partitionKey: PartitionKey) : Operation {
+                val udt = context.session.cluster.getMetadata().getKeyspace(context.session.loggedKeyspace).getUserType("sensor_data_details")
```

Comment:
Instead of doing this on every mutation, can we do this once at the creation of the runner, in order to reduce the overhead of getNextMutation?